context/app/static/js/components/detailPage/Protocol/Protocol.tsx

```diff
@@ -23,32 +38,46 @@ function ProtocolMessage({ precedingText }) {
   );
 }
 
-function ProtocolLink({ url, index }) {
+interface ProtocolLinkProps {
+  url: string;
+  index: number;
+}
+
+function ProtocolLink({ url, index }: ProtocolLinkProps) {
   const { isLoading, data, error } = useProtocolData(url);
 
-  if (error || isLoading || !data) {
+  const trackEntityPageEvent = useTrackEntityPageEvent();
+  const hubmapId = useFlaskDataContext().entity.hubmap_id;
+
+  if (isLoading) {
     if (index !== 0) {
       // Only show loading message for first protocol link
       return null;
     }
+    return <ProtocolMessage isLoading />;
+  }
 
-    if (isLoading && !error) {
-      return (
-        <ProtocolMessage precedingText="Protocols are loading. If protocols take a significant time to load, please " />
-      );
-    }
-
-    return <ProtocolMessage precedingText="Failed to retrieve protocols. Please " />;
+  if (error) {
+    return <ProtocolMessage isError />;
   }
 
   return (
     <SectionItem label={data?.payload?.title}>
-      <OutboundIconLink href={data?.payload?.url}>{data?.payload?.url}</OutboundIconLink>
+      <OutboundIconLink
```

Comment:
I believe the assumption was that if it's not loading and there's no error that the data was returned as expected - I've adjusted this so that the iconlink is not rendered if `data.payload` does not explicitly exist.